transformer_engine/pytorch/tensor/float8_blockwise_tensor.py

```diff
@@ -543,19 +543,79 @@ def forward(
         if shape is None:
             return tensor
 
-        if list(shape) != list(tensor.shape):
-            raise NotImplementedError("View not implemented.")
-        return tensor
+        # Canonicalize shape
+        if not isinstance(shape, Iterable):
+            shape = [shape]
+        elif len(shape) == 1 and isinstance(shape[0], Iterable):
+            shape = shape[0]
+        if -1 in shape:
+            shape = list(shape)
+            d_inferred = -math.prod(ctx.shape) // math.prod(shape)
+            for i, d in enumerate(shape):
+                if d == -1:
+                    shape[i] = d_inferred
+                    break
+        if shape[-1] != ctx.shape[-1]:
```

Comment:
We assume here that `ctx.shape` is available, but earlier we have logic to handle `ctx=None`:
https://github.com/NVIDIA/TransformerEngine/blob/402b270bf8f6b7a6f445ab67e43055aca2f31f19/transformer_engine/pytorch/tensor/float8_blockwise_tensor.py#L541

I think we should just assume that `ctx` is a valid context object and stop supporting the `ctx=None` case. It's nonstandard, we never take advantage of that case (perhaps the original use-case was removed?), and `MXFP8Tensor` and `Float8Tensor` don't support it. We should also make a similar change to `_ReshapeFunc`.